openstudiocore/src/model/ThermalZone.cpp

```diff
@@ -998,14 +993,32 @@ namespace detail {
 
   bool ThermalZone_Impl::setThermostat(const Thermostat & thermostat)
   {
-    resetThermostat();
-    return setPointer(OS_ThermalZoneFields::ThermostatName, thermostat.handle());
+    auto t_model = model();
+
+    if( t_model != thermostat.model() ) {
```

Comment:
This check seems ridiculous but honestly I've considered it, it is legit error checking.  I guess normally this is included if you check the return value of set methods but this could save you headache to do it first.